src/kernel.hpp

```diff
@@ -350,6 +346,16 @@ struct cvk_kernel_argument_values {
             if (resource)
                 resource->release();
         }
+        std::lock_guard<std::mutex> lock(m_lock);
+        if (--m_descriptor_sets_refcount == 0) {
+            m_is_enqueued = false;
+            for (auto ds : m_descriptor_sets) {
```

Comment:
```suggestion
            for (auto& ds : m_descriptor_sets) {
```

I think you intended `ds` to be a reference.

Reply:
Thanks!
I'll add back the destructor as this bug was making it segfault, but I think we should really keep the destructor.